gst/isomp4/qtdemux.c

```diff
@@ -6527,6 +6528,19 @@ qtdemux_parse_samples (GstQTDemux * qtdemux, QtDemuxStream * stream, guint32 n)
         stream->ctts_soffset =
             gst_byte_reader_get_int32_be_unchecked (&stream->ctts);
         stream->ctts_sample_index = 0;
+
+        /* If the ctts offset of a keyframe is not 0, we need to normalize 
+         * ctts offsets so that key frames have a 0 ctts offset.
+         * Otherwise, an unintended time shift will be applied. 
+         */
+        if (cur->keyframe) {
```

Comment:
Shouldn't this logic only be applied for the first keyframe ?

Reply:
I do not think so. Regardless of the way ccts offset is encoded, it is the same for all keyframes. If eventually a keyframe has a different value, it can only be because the normalization kind changed and we should update the error offset accordingly.

Reply:
There is something I still don't understand in this algorithm, shouldn't the offset of the first keyframe applied to all samples?